examples/mnit_ballz/src/objects.nit

```diff
@@ -0,0 +1,90 @@
+# Module containing all objects of the game
+module objects
+
+import geometry
+import geometry::polygon
+import geometry::boxes
+import collision
+
+# The ball is the main character of the game !
+class Ball
+	#center of the ball
+	var center: Point[Float] is writable
+end
+
+# Walls make the ball bounce on them
+class Wall
+	# coordinates of the center of the wall
+	var center: Point[Float]
+
+	# angle in degrees
+	var angle: Float
+
+	# Scale for drawing `self`
+	var scale: Float is noautoinit
+
+	# Width of `self`
+	var width: Float is noautoinit
+
+	# Height of `self`
+	var height: Float is noautoinit
+
+	# Lines composing `self`
+	var lines: Array[OrientedLine[Float]] is noautoinit
+
+	# Initialize `self` with all its lines from `center` and `angle`
+	init complete(center: Point[Float], angle: Float,  scale: Float) do
+		self.center = center
+		self.scale = scale
+		self.height = 128.0
+		self.width = 32.0
+		self.angle = angle
+		var i = new Point[Float](0.0, 0.0)
+		var j = new Point[Float](0.0, 0.0)
+
+		var a = new Point[Float]((center.x - width/2.0), center.y)
+		var b = new Point[Float]((center.x), (center.y - height/2.0))
+		var c = new Point[Float]((center.x + width/2.0), center.y)
+		var d = new Point[Float]((center.x), (center.y + height/2.0))
+
+
+		var l1 = new OrientedLine[Float](i, j, angle - pi/2.0, height * scale.to_f, rotate_point(a, center, angle))
+		var l2 = new OrientedLine[Float](i, j, angle, width * scale.to_f, rotate_point(b, center, angle))
+		var l3 = new OrientedLine[Float](i, j, angle - pi/2.0, height * scale.to_f, rotate_point(c, center, angle))
+		var l4 = new OrientedLine[Float](i, j, angle, width * scale.to_f, rotate_point(d, center, angle))
+		lines = new Array[OrientedLine[Float]]
+		lines.add_all([l1, l2, l3, l4])
+	end
+end
+
+# A line represented with a center and an angle
+class OrientedLine[N]
+	super Line[Numeric]
+	redef type P: Point[Float]
```

Comment:
`N` is unused, and the conjoint use of Numeric and Float make me feel uneasy. please stop doing that.